sql/core/src/test/scala/org/apache/spark/sql/DataFrameNaFunctionsSuite.scala

```diff
@@ -194,4 +194,45 @@ class DataFrameNaFunctionsSuite extends QueryTest with SharedSQLContext {
     assert(out1(4) === Row("Amy", null, null))
     assert(out1(5) === Row(null, null, null))
   }
+
+  test("Spark-12231: dropna with partitionBy and groupBy") {
```

Comment:
You are right, this problem is not related to na.drop. At that time, I was not sure where I can put the test case, so I just keep here. Thanks for the suggestion about putting with the other DataSource tests, I will change the test case and look for the place around the DataSource tests. Thanks very much !